lua/lspconfig/vala_ls.lua

```diff
@@ -5,9 +5,25 @@ configs.vala_ls = {
   default_config = {
     cmd = {'vala-language-server'},
     filetypes = {'vala', 'genie'},
-    root_dir = function(fname)
-      return util.root_pattern("meson.build")(fname)
-        or util.find_git_ancestor(fname)
+    root_dir = function (fname)
+      local root = util.search_ancestors(fname, function (path)
+        local pattern = "meson.build"
+        local f = vim.fn.glob(util.path.join(path, pattern))
+        if not util.path.exists(f) then
+          return nil
+        end
+        for line in io.lines(f) do
+          local str = line:gsub('%s+', '')
+          if str ~= '' then
+            if str:match('^project%s*%(') then
```

Comment:
I'm not at all familiar with Vala. But is there no CLI command that can be used instead? For example, for Rust you can do `cargo metadata` to get the workspace/project root.